retrieval/target.go

```diff
@@ -204,7 +207,46 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 
 	t.scrapeInterval = time.Duration(cfg.ScrapeInterval)
 	t.deadline = time.Duration(cfg.ScrapeTimeout)
-	t.httpClient = httputil.NewDeadlineClient(time.Duration(cfg.ScrapeTimeout))
```

Comment:
Given the client setup complexity this might be better factored out so it doesn't shadow the fundamental logic of the `Update()` method. This also makes it easier to test.

Some comments along the way would certainly help, too.